mephisto/tools/data_browser.py

```diff
@@ -28,45 +25,47 @@ def __init__(self, db=None):
             db = LocalMephistoDB()
         self.db = db
 
-    def _get_units_for_task_runs(self, task_runs: List[TaskRun]) -> List[Unit]:
+    def collect_matching_units_from_task_runs(
+        self, task_runs: List[TaskRun], statuses: List[str]
+    ) -> List[Unit]:
         """
-        Return a list of all Units in a terminal completed state from all
-        the provided TaskRuns.
+        Loops through task_runs to collect all units in the provided statuses list
         """
         units = []
         for task_run in task_runs:
             assignments = task_run.get_assignments()
             for assignment in assignments:
                 found_units = assignment.get_units()
                 for unit in found_units:
-                    if unit.get_status() in [
-                        AssignmentState.COMPLETED,
-                        AssignmentState.ACCEPTED,
-                        AssignmentState.REJECTED,
-                        AssignmentState.SOFT_REJECTED,
-                    ]:
+                    if unit.get_status() in statuses:
                         units.append(unit)
         return units
 
+    def _get_units_for_task_runs(self, task_runs: List[TaskRun]) -> List[Unit]:
+        """
+        Return a list of all Units in a terminal completed state from all
+        the provided TaskRuns.
+        """
+        statuses = [
+            AssignmentState.COMPLETED,
+            AssignmentState.ACCEPTED,
+            AssignmentState.REJECTED,
+            AssignmentState.SOFT_REJECTED,
+        ]
+        return self.collect_matching_units_from_task_runs(task_runs, statuses)
```

Comment:
i think passing in the negation (filtering) would feel better here, e.g. all tasks except those that are EXPIRED, but I'm just nit-picking, this is fine

Reply:
I feel like that would just make the code harder to understand as you would have to go to assignment_state.py to figure out which AssignmentState's are being accepted. 

It wouldn't really make it more concise either as the code would become

```python
filtered_statuses=[
  AssignmentState.CREATED,
  AssignmentState.LAUNCHED,
  AssignmentState.ASSIGNED,
  AssignmentState.MIXED,
  AssignmentState.EXPIRED  
]
```

Reply:
Generally for legibility, it could also be OK to put these collections inside of the `AssignmentState` class as class functions. `AssignmentState.completed()` and `AssignmentState.final_agent()` come to mind as relevant for these two lists at the moment.